pkg/compatibility/compatibility.go

```diff
@@ -0,0 +1,152 @@
+		// update documentation.
+
+		// Full Support
+		"4.10": {
+			GADate:  time.Date(2022, 3, 10, 0, 0, 0, 0, time.UTC), // March 10, 2022
+			FSEDate: time.Date(2022, 9, 10, 0, 0, 0, 0, time.UTC), // September 10, 2022
+			MSEDate: time.Date(2023, 9, 10, 0, 0, 0, 0, time.UTC), // September 10, 2023
+			// Note: FSEDate (Release of 4.11 + 3 months) is currently a "guess".  Update when available.
+		},
+		"4.9": {
+			GADate:  time.Date(2021, 10, 18, 0, 0, 0, 0, time.UTC), // October 18, 2021
+			FSEDate: time.Date(2022, 6, 8, 0, 0, 0, 0, time.UTC),   // June 8, 2022
+			MSEDate: time.Date(2023, 4, 18, 0, 0, 0, 0, time.UTC),  // April 18, 2023
+			// Note: FSEDate (Release of 4.10 + 3 months) is currently a "guess".  Update when available.
+		},
+
+		// Maintenance Support
+		"4.8": {
+			GADate:  time.Date(2021, 7, 27, 0, 0, 0, 0, time.UTC), // July 27, 2021
+			FSEDate: time.Date(2022, 1, 27, 0, 0, 0, 0, time.UTC), // January 27, 2022
+			MSEDate: time.Date(2023, 1, 27, 0, 0, 0, 0, time.UTC), // January 27, 2023
+		},
+		"4.7": {
+			GADate:  time.Date(2021, 2, 24, 0, 0, 0, 0, time.UTC),  // February 24, 2021
+			FSEDate: time.Date(2021, 10, 27, 0, 0, 0, 0, time.UTC), // October 27, 2021
+			MSEDate: time.Date(2022, 8, 24, 0, 0, 0, 0, time.UTC),  // August 24, 2022
+		},
+
+		// End of life
+		"4.6": {
+			GADate:  time.Date(2020, 10, 27, 0, 0, 0, 0, time.UTC), // October 27, 2020
+			FSEDate: time.Date(2021, 3, 24, 0, 0, 0, 0, time.UTC),  // March 24, 2021
+			MSEDate: time.Date(2021, 10, 18, 0, 0, 0, 0, time.UTC), // October 18, 2022
+		},
+		"4.5": {
+			GADate:  time.Date(2020, 7, 13, 0, 0, 0, 0, time.UTC),  // July 13, 2020
+			FSEDate: time.Date(2020, 11, 27, 0, 0, 0, 0, time.UTC), // November 27, 2020
+			MSEDate: time.Date(2021, 7, 27, 0, 0, 0, 0, time.UTC),  // July 27, 2021
+		},
+		"4.4": {
+			GADate:  time.Date(2020, 5, 5, 0, 0, 0, 0, time.UTC),  // May 5, 2020
+			FSEDate: time.Date(2020, 8, 13, 0, 0, 0, 0, time.UTC), // August 13, 2020
+			MSEDate: time.Date(2021, 2, 24, 0, 0, 0, 0, time.UTC), // February 24, 2021
+		},
+		"4.3": {
+			GADate:  time.Date(2020, 1, 23, 0, 0, 0, 0, time.UTC),  // January 23, 2020
+			FSEDate: time.Date(2020, 6, 5, 0, 0, 0, 0, time.UTC),   // June 5, 2020
+			MSEDate: time.Date(2020, 10, 27, 0, 0, 0, 0, time.UTC), // October 27, 2020
+		},
+		"4.2": {
+			GADate:  time.Date(2019, 10, 16, 0, 0, 0, 0, time.UTC), // October 16, 2019
+			FSEDate: time.Date(2020, 2, 23, 0, 0, 0, 0, time.UTC),  // February 23, 2020
+			MSEDate: time.Date(2020, 7, 13, 0, 0, 0, 0, time.UTC),  // July 13, 2020
+		},
+		"4.1": {
+			GADate:  time.Date(2019, 6, 4, 0, 0, 0, 0, time.UTC),   // June 4, 2019
+			FSEDate: time.Date(2019, 11, 16, 0, 0, 0, 0, time.UTC), // November 16, 2019
+			MSEDate: time.Date(2020, 5, 5, 0, 0, 0, 0, time.UTC),   // May 5, 2020
+		},
+	}
+)
+
+func GetLifeCycleDates() map[string]VersionInfo {
+	return ocpLifeCycleDates
+}
+
+func DetermineOCPStatus(version string, date time.Time) string {
```

Comment:
Minor issue: Is it really useful having the date as an argument for this function? You can actually do that `date := time.Now()` inside. I mean, I don't see in which scenario someone may call this function with something different than the current time.

Reply:
I did this because of unit testing.  I wanted to be able to pass in a provided date and see if my function worked.